### ISSUE TITLE
Couldn't run kube bench on azure AKS

### ISSUE THREAD

**Issue by VineethReddy02** [aquasecurity/kube-bench#318]
I am running an AKS cluster precisely a single node cluster. I access this cluster from my local machine by configuring in KUBECONFIG. When I run kube-bench the pods gets into the error state and the logs say that couldn't find the kubectl or kubelet version.
I was successful in running the kube-bench using minikube.
Let me know if am missing anything. Thanks! :)

**Comment by ttousai** [comment]
@VineethReddy02 please add the following information:

```
- Note the version of kube-bench you are running and the command line options you are using. 
- Note the version of kubernetes you are running (from `kubectl version` or `oc version` for Openshift).
- Set `-v 10` command line option and save the log output. Please paste this into your issue.
```

**Comment by lizrice** [comment]
Any chance that #332 would explain it?

**Comment by PatrickSplice** [comment]
My guess is the original poster was trying to run a job using one of the supplied job.yaml files.  AKS clusters install their kubelet and kubectl into non-standard locations on their nodes.  They end up in /usr/local/bin.  You will also find that the kubeconfig file is in a non-standard location.  I tried to fix this by modifying the job.yaml file and mounting /usr/local/bin.  Surprise, this doesn't work because kube-bench is installed in /usr/local/bin.  I also tried mounting the local node's /usr/local/bin to a separate mount-point inside the container and adding an ENV setting for the container to ensure that dir was int he PATH, but that also failed.  I overcame this by using kubectl from our Jenkins pipeline docker container to determine the kubernetes version and injecting it into the job.yaml.  This got the job to run.  The next issue was a whole bunch of false positives because AKS stored many config files, cert files, etc in non-standard locations.  I was able to modify the default ./cfg/config.yaml to include these locations so it would find those files when running tests.

It would be nice if those config.yaml changes could be incorporated into the master branch.

Can I checkout the contributing branch and submit a PR?

**Comment by lizrice** [comment]
Hi @PatrickSplice, absolutely, we’d love a PR! 

If it works to do this, the various file locations would go as new alternatives in the lists in cfg/config.yaml, if possible. We would be totally open to a job-aks.yaml (we already have a job-eks.yaml), which might be needed to change the file mounts.

**Comment by PatrickSplice** [comment]
I was finally able to get a job to run by adding a PATH and KUBECONFIG environment variable and a new mount for the kubectl and kubelet executables.  The issue now is that it fails to determine the kubernetes version and defaults to 1.6.  I'm guessing the reason it can't detect the version is that the kubernetes cert has 600 permissions and is owned by root, so only the root account can run kubectl on the AKS nodes.  I made some half-hearted efforts to see if running the container as root would solve the problem, but didn't spend a lot of time on it.

So, I am back to injecting the version into my job-aks.yaml file as part of my pipeline.  I'll also need to build my own docker image to incorporate the config.yaml.

**Comment by PatrickSplice** [comment]
Pull request 383 created to add config.yaml file updates.

**Comment by lizrice** [comment]
I think this is fixed by #383, but please re-open if that's not the case
